File: app/src/main/java/ch/bailu/aat/description/ContentDescription.java
package ch.bailu.aat.description;

import android.content.Context;

import ch.bailu.aat.dispatcher.OnContentUpdatedInterface;


public abstract class ContentDescription
        implements ContentInterface, OnContentUpdatedInterface {

    public static final String VALUE_DISABLED = "--";

    private final Context context;

    protected static final String NULL_STRING="";
    public ContentDescription(Context c) {
        context = c;
    }

    public Context getContext() {
        return context;
    }


    public abstract String getValue();
    public abstract String getLabel();
    public String getLabelShort() {
        return getLabel();
    }

    public String getUnit() {
        return NULL_STRING;
    }


    @Override
    public String getValueAsString() {
        return getValue() + " " + getUnit();
    }


}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/ContentDescription.kt
package ch.bailu.aat_lib.description

import ch.bailu.aat_lib.dispatcher.TargetInterface

/**
 * Base class for a displayable label+value+unit triple that updates from [GpxInformation].
 *
 * Implements [TargetInterface] so it can be wired into the [Dispatcher] chain
 * (typically wrapped by a [NumberView]). Subclasses override [onContentUpdated]
 * to extract sensor-specific fields from [GpxInformation.getAttributes].
 */
abstract class ContentDescription : ContentInterface, TargetInterface {
    abstract fun getValue(): String

    open fun getLabelShort(): String {
        return getLabel()
    }

    override fun getValueAsString(): String {
        return "${getValue()} ${getUnit()}"
    }

    open fun getUnit(): String {
        return ""
    }

    companion object {
        const val VALUE_DISABLED = "--"
    }
}
